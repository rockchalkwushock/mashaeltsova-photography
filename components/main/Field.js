import { styles } from '../../lib'

const Field = props => {
  if (props.className === 'button') {
    return (
      <div className={props.className}>
        <button type={props.type}>{props.text}</button>
        <style jsx>{`
          .button {
            grid-area: bookingButton;
            grid-column: 1 / 3;
            grid-row: 5 / 6;
          }
          button {
            background-color: ${styles.colors.warm};
            border: 0;
            box-shadow: 4px 4px 4px rgba(54, 69, 79, 0.7);
            color: ${styles.colors.white};
            font-family: ${styles.fonts.courgette};
            font-size: 1em;
            outline: 0;
            padding: 1em;
            text-transform: uppercase;
            width: 100%;
          }
          button:focus,
          button:hover {
            background-color: ${styles.colors.powder};
            color: ${styles.colors.warm};
            font-size: 1.2em;
            -webkit-transition: background-color 0.3s ease-out;
            -moz-transition: background-color 0.3s ease-out;
            transition: background-color 0.3s ease-out;
          }
          @media (min-width: 768px) {
            .button {
              grid-area: bookingButton;
              grid-column: 1 / 3;
              grid-row: 4 / 5;
            }
          }
        `}</style>
      </div>
    )
  }
  return (
    <div className={props.className}>
      <label htmlFor={props.className}>{props.text}</label>
      <input type={props.type} name={props.className} />
      <style jsx>{`
        label {
          display: block;
        }
        input {
          border: 0;
          border-bottom: 1px solid ${styles.colors.warm};
          color: ${styles.colors.charcoal};
          outline: 0;
          padding: 0 0.5em;
          width: 100%;
        }
        .firstName {
          grid-area: firstName;
          grid-column: 1 / 2;
          grid-row: 1 / 2;
        }
        .lastName {
          grid-area: lastName;
          grid-column: 2 / 3;
          grid-row: 1 / 2;
        }
        .email {
          grid-area: email;
          grid-column: 1 / 3;
          grid-row: 2 / 3;
        }
        .phone {
          grid-area: phone;
          grid-column: 1 / 3;
          grid-row: 3 / 4;
        }
        .message {
          grid-area: message;
          grid-column: 1 / 3;
          grid-row: 4 / 5;
        }
        @media (min-width: 768px) {
          .email {
            grid-area: email;
            grid-column: 1 / 2;
            grid-row: 2 / 3;
          }
          .phone {
            grid-area: phone;
            grid-column: 2 / 3;
            grid-row: 2 / 3;
          }
          .message {
            grid-area: message;
            grid-column: 1 / 3;
            grid-row: 3 / 4;
          }
        }
      `}</style>
    </div>
  )
}

export default Field
