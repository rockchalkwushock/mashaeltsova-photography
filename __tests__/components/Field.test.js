import React from 'react'
import { render } from 'enzyme'

import Field from '../../components/main/Field'

describe('Component: Field', () => {
  test('Renders without exploding as .firstName', () => {
    const tree = render(
      <Field className="firstName" text="First Name" type="text" />
    )
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .lastName', () => {
    const tree = render(
      <Field className="lastName" text="Last Name" type="text" />
    )
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .email', () => {
    const tree = render(<Field className="email" text="Email" type="email" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .phone', () => {
    const tree = render(<Field className="phone" text="Phone" type="tel" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .message', () => {
    const tree = render(
      <Field className="message" text="Short Message" type="text" />
    )
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .button', () => {
    const tree = render(
      <Field className="button" text="Book Now!" type="submit" />
    )
    expect(tree).toMatchSnapshot()
  })
})
