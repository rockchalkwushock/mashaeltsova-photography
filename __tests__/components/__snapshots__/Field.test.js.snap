// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Field Renders without exploding as .button 1`] = `
<div
  class="jsx-3953151939 button"
>
  <button
    class="jsx-3953151939"
    type="submit"
  >
    Book Now!
  </button>
</div>
`;

exports[`Component: Field Renders without exploding as .email 1`] = `
<div
  class="jsx-3773761155 email"
>
  <label
    class="jsx-3773761155"
    for="email"
  >
    Email
  </label>
  <input
    class="jsx-3773761155"
    name="email"
    type="email"
  />
</div>
`;

exports[`Component: Field Renders without exploding as .firstName 1`] = `
<div
  class="jsx-3773761155 firstName"
>
  <label
    class="jsx-3773761155"
    for="firstName"
  >
    First Name
  </label>
  <input
    class="jsx-3773761155"
    name="firstName"
    type="text"
  />
</div>
`;

exports[`Component: Field Renders without exploding as .lastName 1`] = `
<div
  class="jsx-3773761155 lastName"
>
  <label
    class="jsx-3773761155"
    for="lastName"
  >
    Last Name
  </label>
  <input
    class="jsx-3773761155"
    name="lastName"
    type="text"
  />
</div>
`;

exports[`Component: Field Renders without exploding as .message 1`] = `
<div
  class="jsx-3773761155 message"
>
  <label
    class="jsx-3773761155"
    for="message"
  >
    Short Message
  </label>
  <input
    class="jsx-3773761155"
    name="message"
    type="text"
  />
</div>
`;

exports[`Component: Field Renders without exploding as .phone 1`] = `
<div
  class="jsx-3773761155 phone"
>
  <label
    class="jsx-3773761155"
    for="phone"
  >
    Phone
  </label>
  <input
    class="jsx-3773761155"
    name="phone"
    type="tel"
  />
</div>
`;
